kernel/kernel-api/src/test/scala/io/delta/kernel/internal/util/SchemaUtilsSuite.scala

```diff
@@ -310,6 +311,140 @@ class SchemaUtilsSuite extends AnyFunSuite {
+  }
+
+  test("Compute schema changes with dropped fields") {
+    val fieldMappingBefore = Map(
+      1L -> new StructField("id", IntegerType.INTEGER, true),
+      2L -> new StructField("data", StringType.STRING, true)).map { case (k, v) =>
+      java.lang.Long.valueOf(k) -> v
+    }.asJava
+
+    val fieldMappingAfter = Map(
+      2L -> new StructField("data", StringType.STRING, true)).map { case (k, v) =>
+      java.lang.Long.valueOf(k) -> v
+    }.asJava
+
+    val schemaChanges = SchemaUtils.computeSchemaChangesById(fieldMappingBefore, fieldMappingAfter)
+
+    assert(schemaChanges.addedFields().isEmpty)
+    assert(schemaChanges.updatedFields().isEmpty)
+    assert(schemaChanges.removedFields().size() == 1)
+    assert(schemaChanges.removedFields().get(0) == fieldMappingBefore.get(1))
+  }
+
+  test("Compute schema changes with nullability change") {
+    val fieldMappingBefore = Map(
+      1L -> new StructField("id", IntegerType.INTEGER, true),
+      2L -> new StructField("data", StringType.STRING, true)).map { case (k, v) =>
+      java.lang.Long.valueOf(k) -> v
+    }.asJava
+
+    val fieldMappingAfter = Map(
+      1L -> new StructField("id", IntegerType.INTEGER, true),
+      2L -> new StructField("required_data", StringType.STRING, false)).map { case (k, v) =>
+      java.lang.Long.valueOf(k) -> v
+    }.asJava
+
+    val schemaChanges = SchemaUtils.computeSchemaChangesById(fieldMappingBefore, fieldMappingAfter)
+
+    assert(schemaChanges.addedFields().isEmpty)
+    assert(schemaChanges.removedFields().isEmpty)
+    assert(schemaChanges.updatedFields().size() == 1)
+    assert(schemaChanges.updatedFields().get(0) == new Tuple2(
+      fieldMappingBefore.get(2),
+      fieldMappingAfter.get(2)))
+  }
+
+  test("Compute schema changes with moved fields") {
+    val fieldMappingBefore = Map(
+      1L -> new StructField(
+        "struct",
+        new StructType()
+          .add(new StructField("id", IntegerType.INTEGER, true))
+          .add(new StructField("data", StringType.STRING, true)),
+        true)).map { case (k, v) =>
+      java.lang.Long.valueOf(k) -> v
+    }.asJava
+
+    val fieldMappingAfter = Map(
+      1L -> new StructField(
+        "struct",
+        new StructType()
+          .add(new StructField("data", StringType.STRING, true))
+          .add(new StructField("id", IntegerType.INTEGER, true)),
+        true)).map { case (k, v) =>
+      java.lang.Long.valueOf(k) -> v
+    }.asJava
+
+    val schemaChanges = SchemaUtils.computeSchemaChangesById(fieldMappingBefore, fieldMappingAfter)
+
+    assert(schemaChanges.addedFields().isEmpty)
+    assert(schemaChanges.removedFields().isEmpty)
+    assert(schemaChanges.updatedFields().size() == 1)
+    assert(schemaChanges.updatedFields().get(0) == new Tuple2(
+      fieldMappingBefore.get(1),
+      fieldMappingAfter.get(1)))
+  }
+
   ///////////////////////////////////////////////////////////////////////////
   // filterRecursively
```

Comment:
can you add a header like this one for these tests (for computeSchemaChangesById)